### ISSUE TITLE
The original dataset (https://www.ofetnow.top/) contains many mistakes

### ISSUE THREAD

**Issue by Swordshinehjy** [wonderingbiggy/OFETnow#1]
For example, same HOMO/LUMO values for a polymer, incorrect use of Mean + std to obtain the maximum mobility, DOI errors caused by Excel's autofill feature (the last four entries)
These issues should be corrected.

**Comment by Swordshinehjy** [comment]
In some SMILES, missing or unpaired slashes or backslashes used to express double bond configurations lead to the loss of double bond stereochemical information.

**Comment by wonderingbiggy** [comment]
Thank you for reviewing the database and bringing these issues to our attention.

**HOMO/LUMO values.** We have identified entries associated with two publications in which the HOMO and LUMO values were incorrectly recorded as identical. We will correct these values against the original sources before the formal release. Neither HOMO nor LUMO was included in the model features, so these errors do not affect the model training, predictions, or reported model performance.

**Mobility values.** We agree that a mean plus one standard deviation should not, by itself, be treated as an experimentally observed maximum mobility. We will review the relevant entries against the source papers and correct any cases where these quantities were conflated.

**DOI errors.** Some DOI errors originated from spreadsheet autofill during data collection. We previously corrected a number of affected entries, but some remain. We will correct the remaining errors and recheck the DOI fields before the formal release.

**Double-bond stereochemistry in SMILES.** Not every double bond requires an explicit E/Z specification, and an unspecified configuration is not necessarily an incorrect structure. However, where a stereogenic double bond has a clearly specified configuration in the source publication, that information should be preserved. We will review the flagged SMILES for missing or inconsistent directional bond notation and correct them where appropriate, without assigning configurations unsupported by the source.

If possible, please provide the entry IDs for the mobility and SMILES examples to help us locate the specific records. Thank you again for helping improve the database.

**Comment by wonderingbiggy** [comment]
Following a careful review, we have identified five entries affected by the DOI autofill issue and seven records affected by the identical HOMO/LUMO values. We will correct these entries in both the downloadable database file and the website in the next update. Neither HOMO nor LUMO was included in the model features, so the errors in these fields do not affect the model results. We are continuing to investigate the remaining concerns, including mobility values and SMILES stereochemistry. Our review so far suggests that most maximum-mobility entries were taken from values explicitly reported as maxima by the original authors, rather than calculated as the mean plus a standard deviation. Any confirmed errors will be corrected in both the database file and the website. Thank you again for your feedback.

**Comment by Swordshinehjy** [comment]
Please don't use AI to respond. That's not a good habbit. The whole code is typically AI-generated. Although this's a growing trend, this one is an academic paper, after all, and at least requires a review by human experts. The codebase seems to be missing a substantial amount of critical information, including preprocessing steps such as feature selection, missing value imputation, and scaling. I suggest you add such codes. In my view, the current code already signals several potential data leakage issues, including feature selection leakage, BO hyperparameter tuning leakage, and preprocessing on the full dataset. As a paper, all essential code should be fully provided.

For example, it appears to use cross-row statistics derived from the entire dataset rather than being performed after splitting, which is a dangerous practice. Although data scaling has no impact on XGBoost, filling missing values directly affects tree splits. I believe it is necessary to have a human expert thoroughly check your code, or use a stronger AI model to re-evaluate the code quality. If data leakage leads to overly optimistic results, it is much better to refine everything before publication rather than corrections or even retractions. I recently heard of an incident in your school, which are truly unfortunate. Of course, I only browsed this repository and haven't read the paper itself, so I cannot confirm this with absolute certainty.

**Comment by Swordshinehjy** [comment]
It's actually strange that the model features do not include HOMO or LUMO values, given that humans often use these two features for quick screening. Energy levels are directly related to charge injection. Of course, you may have your own considerations. For instance, HOMO/LUMO values in literature come from noisy and messy sources, such as UPS/IPES vs. CV, or even crude approximations where someone ignores exciton binding energy and simply estimate LUMO by adding the optical gap to HOMO. Such data introduce serious noise and make the results unreliable.

In the current version (Sep/16/2026), quite a few double bonds in isoindigo and its analogues are still not handled correctly. RDKit automatically ignores unpaired forward slashes (/) or backslashes (\) without throwing errors down the line, but it treats them as coexisting cis/trans isomers. Other programs may not handle it the same way. I don't know if you directly used an OCSR tool (DECIMER?), but this type of error likely requires additional code checking. A script to check whether all slashes or backslashes in the SMILES strings are paired is recommended.

Tree-based models like XGBoost provide native support for sparse matrices due to their underlying implementation, so feature selection on Morgan fingerprints should be used with caution. Generally, dimensionality reduction is unnecessary for such sparse matrices when using XGB. In fact, I personally advise against using Morgan fingerprints altogether. They are inherently highly redundant and correlated, especially when using a larger radius, which causes smaller-radius substructures to appear simultaneously and leads to highly collinear, redundant information.

**Comment by Swordshinehjy** [comment]
XGB natively supports nan values by only iterating over non-missing samples and temporarily excluding such missing values. It seems you are filling missing values with 0, which leads to information pollution, interferes with its tree-splitting logic, and can even introduce bias.

**Comment by wonderingbiggy** [comment]
Thanks for the comments. Just to clarify a few points:
Regarding the zero values ​​in the feature matrix, most do not represent missing values ​​filled with zeros. These zeros primarily arise from 2048-bit Morgan fingerprints and one-hot encoded categorical features, where zero is the standard representation. The "fillna(0)" applied to the RDKit descriptors was a defensive measure to prevent RDKit from returning invalid numerical values; it was not intended as a general imputation strategy for experimental variables. We have verified this and found no instances of missing RDKit values ​​being systematically converted to zeros.
For the SMILES stereochemistry issue, this was a fair point. We went back and re-checked the relevant structures, added an additional procedure to handle the stereochemical representation of some double bonds more carefully, and have updated the corresponding database files in the repository.
Regarding multiple entries for the same polymer, this is defenitely not a data-augmentation strategy. Each row represents an actual device entry reported under a specific set of processing and device conditions. In OFETs, the same polymer can give quite different transport behavior under different conditions—for example, annealing alone can change the mobility class and changing of the device architecture can lead to the change of the type of carriers, this is observed in our experiments frequently. So these entries are treated as different devices rather than duplicated samples of the same polymer.
As for feature selection and hyperparameter tuning, their overall effect on the final performance is relatively limited. Feature selection was mainly used to reduce redundant/noisy variables and make the model more compact and easier to interpret, rather than to create the predictive signal itself. Hyperparameter tuning similarly provides refinement rather than a fundamental change in model behavior. this part you can refre to our supporting informations for the detail.

**Comment by wonderingbiggy** [comment]
Nearly all the structures were collected manually. The lack of stereochemistry may due to the program we use to convert ChemDraw structures into SMILES strings.

**Comment by Swordshinehjy** [comment]
Most part of the response is still AI-generated. Only the last sentence  is hand-writing. 

Let me be clear. I pointed out the potential issues of data leakage, specifically, feature selection leakage, BO hyperparameter tuning leakage, and preprocessing, cuz the current code indicates that the full dataset has been scaled and the parameter has been given before data splitting, Your response is not a rebuttal but a non sequitur. If test information enters feature selection, preprocessing, or hyperparameter search, the reported performance is invalid. The only relevant question is: what data touched what step, and when? You should provide the code.

And I did not mention the RDKIT's internal strategies to fill nan values. I mean that Capacitance, Aspect_Ratio, Annealing_Temperature, and PDI in the file [training_features.csv](https://github.com/wonderingbiggy/OFETnow/blob/main/data/training_features.csv) may have been improperly imputed. 

You said that you collected the structures using Chemdraw and converted them using another program. Please note that Alt+Ctrl+C can directly convert a structure to a SMILES string in Chemdraw.

**Comment by Swordshinehjy** [comment]
As you mentioned "multiple entries for the same polymer", this is not in my previously raised issues. Glad to know that your AI itself detected such scaffold leakage. I think scaffold leakage is sometimes unavoidable for a small dataset, but the effect can be readily quantified by comparing StratifiedKFold with StratifiedGroupKFold (groups=canonical SMILES). Also, you have introduced very long side chains in a co-repeating unit, which may dilute backbone signal, so a more advanced group split is to use Murcko scaffold.

And here is my result: "You may possibly UNDERESTIMATE the effect of scaffold leakage in this project."

The released protocol splits device rows with a random StratifiedKFold, but the rows are not independent: one polymer appears on many rows (one row per device). A test row whose polymer also occurs in the training fold is not an unseen compound, so random CV measures interpolate a new device of a known polymer, not predict a new polymer. 

I used a script to measure that gap. Only the split changes. The data, the features, the XGBoost hyperparameters and the seeds are identical between protocols (features rebuilt from the raw export, one-hot refit inside each training fold, no imputation / no scaling / no feature selection, averaged from three random seeds):

  random          StratifiedKFold(shuffle=True)                  - released protocol
  group_molecule  StratifiedGroupKFold(groups=canonical SMILES)  - no polymer shared
  group_scaffold  StratifiedGroupKFold(groups=Murcko scaffold)   - no backbone shared

| task	| random | group_molecule | group_scaffold | 
|---|---|---|---|
| mobility acc	| 0.8205	| 0.7722 (−4.8 pt) | 	0.7072 (−11.3 pt) | 
| carrier-type acc	| 0.9247	| 0.8462 (−7.9 pt)	| 0.7787 (−14.6 pt) | 

>  Regarding multiple entries for the same polymer, this is defenitely not a data-augmentation strategy. Each row represents an actual device entry reported under a specific set of processing and device conditions. In OFETs, the same polymer can give quite different transport behavior under different conditions—for example, annealing alone can change the mobility class and changing of the device architecture can lead to the change of the type of carriers, this is observed in our experiments frequently. So these entries are treated as different devices rather than duplicated samples of the same polymer.

**Comment by Swordshinehjy** [comment]
Using Morgan fingerprints with the radius=8 is a critical issue. This setting leads to serious hash collisions. A typical setting is radius = 2 or 3. If differents structures share the same fingerprint, you may simply concat extra descriptors for side-chain length, branching point, etc.

Moreover, using Morgan fingerprints for polymer usually require a few technically tricky operations. Because you only gives the monomer structures, the monomer truncation inevitably introduces end-group artifacts. A common method is to extend a monomer to an oligomer (trimer or 5-mer, sometimes even cyclizing the oligomer) before computing the Morgan fingerprints.
